tests/test_modeling_wav2vec2.py

```diff
@@ -1495,6 +1496,30 @@ def test_wav2vec2_with_lm(self):
 
         self.assertEqual(transcription[0], "bien y qué regalo vas a abrir primero")
 
+    @require_pyctcdecode
+    @require_torchaudio
+    def test_wav2vec2_with_local_lm(self):
+        local_dir = snapshot_download("hf-internal-testing/processor_with_lm")
+
+        ds = load_dataset("common_voice", "es", split="test", streaming=True)
+        sample = next(iter(ds))
+
+        resampled_audio = torchaudio.functional.resample(
+            torch.tensor(sample["audio"]["array"]), 48_000, 16_000
+        ).numpy()
+
+        model = Wav2Vec2ForCTC.from_pretrained(local_dir).to(torch_device)
+        processor = Wav2Vec2ProcessorWithLM.from_pretrained(local_dir)
+
+        input_values = processor(resampled_audio, return_tensors="pt").input_values
+
+        with torch.no_grad():
+            logits = model(input_values.to(torch_device)).logits
+
+        transcription = processor.batch_decode(logits.cpu().numpy()).text
+
+        self.assertEqual(transcription[0], "bien y qué regalo vas a abrir primero")
+
```

Comment:
This is not needed, since you've added the test directly to the processor (which is better IMO)

Reply:
Will remove it then since it's testing twice the same thing.